### PR TITLE
Optimize rockcraft build

### PR DESCRIPTION
**Pull request by maci3jka**
Currently, for each change in the rockcraft repo, we publish the current state of rockcraft files and run tests on them even if files haven’t changed. We need to build only changed files but run tests on all rock images.

The PR updates pushed images after the build to only those modified rocks and also, bump the rockcraft version which fixes the CI build.

KU-1195

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -8,16 +8,22 @@ on:
 jobs:
   build-and-push-arch-specifics:
     name: Build Rocks and Push Arch Specific Images
-    uses: canonical/k8s-workflows/.github/workflows/build_rocks.yaml@main
+    uses: canonical/k8s-workflows/.github/workflows/build_rocks.yaml@story/KU-1195/fix-when-image-not-present
```
Comment (bschimke95) [inline .github/workflows/pull_request.yaml:11]: this should be on main

File: .gitignore
Comment (bschimke95) [inline .gitignore:1]: this file and below are not related to the PR description

**Comment by maci3jka** [comment]
merge #14 before

File: .gitignore
Comment (maci3jka) [reply .gitignore:1]: moved into #14

**Review by bschimke95 — COMMENTED** [review]
`run-tests` need to be moved as well. Otherwise the CI fails.

**Review by bschimke95 — APPROVED** [review]
LGTM
